nemo/nemo/core/neural_factory.py

```diff
@@ -646,6 +646,23 @@ def deployment_export(self,
             input_example: sometimes tracing will require input examples
             output_example: Should match inference on input_example
         """
+        # Custom hacks
+        # We are checking type like this to avoid taking dependency on nemo_asr
+        if type(module).__name__ == "JasperEncoder":
```

Comment:
Can we add this code to the `JasperEncoder` class somehow and have this be a universal check? Like a `prepare_for_deployment()` abstract method on TrainableNeuralModule that may or may not be implemented?

Seems less than ideal for this to be in the neural_factory code.

Reply:
good idea, but I would say this is outside of this PR, since we'd want to do it properly

Reply:
Yes, but a step in the right direction isn't much different -- 

```python
if module.hasattr("prepare_for_deployment"):
    module.prepare_for_deployment()
```

then the encoder has this code pretty much copy pasted to it, with `module` being replaced with `self`

Reply:
We can build upon this in future releases, but at least that way we don't have more model-specific code in a place that should be demonstrably model-agnostic.

Reply:
let's do this properly for >0.9.0